tests/test_posixtestsuite.py

```diff
@@ -0,0 +1,3423 @@
+                   '-s', 'PTHREAD_POOL_SIZE=21',
+                   '-I', posix_test_dir('pthread_create')])
+
+
+  @requires_threads
+  def test_pthread_create_1_6(self):
+    self.btest(posix_test_file('pthread_create', '1-6.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=34',
+                   '-I', posix_test_dir('pthread_create')])
+
+
+# Test throws exception: need to fix thread function signature: "void*()" -> "void*(void*)"
+#  @requires_threads
+#  def test_pthread_create_2_1(self):
+#    self.btest(posix_test_file('pthread_create', '2-1.c'),
+#               expected = '0',
+#               args = posix_tests_common_args + [
+#                   '-s', 'USE_PTHREADS=1',
+#                   '-s', 'PTHREAD_POOL_SIZE=8',
+#                   '-I', posix_test_dir('pthread_create')])
+
+
+# Test throws exception and hangs: need to fix thread function signature: "void*()" -> "void*(void*)"
+#  @requires_threads
+#  def test_pthread_create_3_1(self):
+#    self.btest(posix_test_file('pthread_create', '3-1.c'),
+#               expected = '0',
+#               args = posix_tests_common_args + [
+#                   '-s', 'USE_PTHREADS=1',
+#                   '-s', 'PTHREAD_POOL_SIZE=8',
+#                   '-I', posix_test_dir('pthread_create')])
+
+
+  @requires_threads
+  def test_pthread_create_3_2(self):
+    self.btest(posix_test_file('pthread_create', '3-2.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_create')])
+
+
+# Test fails: need to fix thread function signature: "void*()" -> "void*(void*)"
+#  @requires_threads
+#  def test_pthread_create_4_1(self):
+#    self.btest(posix_test_file('pthread_create', '4-1.c'),
+#               expected = '0',
+#               args = posix_tests_common_args + [
+#                   '-s', 'USE_PTHREADS=1',
+#                   '-s', 'PTHREAD_POOL_SIZE=8',
+#                   '-I', posix_test_dir('pthread_create')])
+
+
+  @requires_threads
+  def test_pthread_create_5_1(self):
+    self.btest(posix_test_file('pthread_create', '5-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_create')])
+
+
+  @requires_threads
+  def test_pthread_create_5_2(self):
+    self.btest(posix_test_file('pthread_create', '5-2.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_create')])
+
+
+# Test throws exception: need to fix thread function si
```

Comment:
There is soo much copy-paste in this file. :( The original test suite enumerated files in directories to run all tests in the suite, can we follow that approach to avoid hand-maintaining this file for each test?

Reply:
In 4c4fd41 I've refactored this to generate test methods based on list of valid Test Cases.

My idea is to have all valid TCs (valid in browser environment) listed in this file. In this approach we avoid running TCs which won't work in with Emscripten (i.e. ones testing fork and signals). WDYT?